sdk/communication/azure-communication-chat/README.md

```diff
@@ -0,0 +1,712 @@
+    public void onSuccess(MultiStatusResponse result, okhttp3.Response response) {
+        // MultiStatusResponse is the result returned from creating a thread.
+        // It has a 'multipleStatus' property which represents a list of IndividualStatusResponse.
+        String threadId;
+        List<IndividualStatusResponse> statusList = result.getMultipleStatus();
+        for (IndividualStatusResponse  status : statusList) {
+            if (status.getId().endsWith("@thread.v2")
+                && status.getType().contentEquals("Thread")) {
+                threadId = status.getId();
+                break;
+            }
+        }
+        // TODO: Take further action.
+    }
+
+    public void onFailure(Throwable throwable, okhttp3.Response response) {
+        // TODO: Display error message.
+    }
+});
+```
+
+### Get a thread
+
+Use the `getChatThread` method to retrieve a thread.
+```java
+// The unique ID of the thread.
+final String threadId = "<thread_id>";
+client.getChatThread(threadId, new Callback<ChatThread>() {
+    @Override
+    public void onSuccess(ChatThread thread, Response response) {
+        //TODO: Take further action
+    }
+
+    @Override
+    public void onFailure(Throwable throwable, Response response) {
+        // TODO: Display error message.
+    }
+});
+```
+
+### List threads
+
+Use the `listChatThreads` method to retrieve a list of threads.
+
+```java
+// The maximum number of messages to be returned per page, optional.
+final int maxPageSize = 10;
+// The thread start time to consider in the query, optional
+final OffsetDateTime startTime = OffsetDateTime.parse("2020-09-08T00:00:00Z");
+client.listChatThreadsPages(maxPageSize, startTime,
+    new Callback<AsyncPagedDataCollection<ChatThreadInfo, Page<ChatThreadInfo>>>() {
+    @Override
+    public void onSuccess(AsyncPagedDataCollection<ChatThreadInfo, Page<ChatThreadInfo>> pageCollection,
+                          Response response) {
+     // pageCollection enables enumerating list of threads.                       
+     pageCollection.getFirstPage(new Callback<Page<ChatThreadInfo>>() {
+            @Override
+            public void onSuccess(Page<ChatThreadInfo> firstPage, Response response) {
+                for (ChatThreadInfo thread : firstPage.getItems()) {
+                    //TODO: Take further action
+                }
+                retrieveNextThreadPages(firstPage.getPageId(), pageCollection);
+            }
+
+            @Override
+            public void onFailure(Throwable throwable, Response response) {
+                // TODO: Display error message.
+            }
+        });
+    }
+
+    @Override
+    public void onFailure(Throwable throwable, Response response) {
+        // TODO: Display error message.
+    }
+});
+
+void retrieveNextThreadPages(String nextPageId,
+                       AsyncPagedDataCollection<ChatThreadInfo, Page<ChatThreadInfo>> pageCollection) {
```

Comment:
The indentation looks strange in the web viewer, is this how it looks in an IDE?

Reply:
when we render it, it looks correct, (on my mac safari)

<img width="865" alt="Screen Shot 2020-10-02 at 3 55 51 PM" src="https://user-images.githubusercontent.com/1471612/94975907-017ef580-04c8-11eb-9552-f0bbe17c89a1.png">

Reply:
indentation is updated across samples